src/qr.jl

```diff
@@ -538,8 +538,10 @@ end
 function rank(A::QRPivoted; atol::Real=0, rtol::Real=min(size(A)...) * eps(real(float(one(eltype(A.Q))))) * iszero(atol))
     m = min(size(A)...)
     m == 0 && return 0
-    tol = max(atol, rtol*abs(A.R[1,1]))
-    return something(findfirst(i -> abs(A.R[i,i]) <= tol, 1:m), m+1) - 1
+    rdiag = diag(getfield(A, :factors))
```

Comment:
This makes an unnecessary copy.  Maybe just set `factors = getfield(A, :factors)` and work directly with `factors[i,i]`?

Reply:
i.e. just replace `A.R` with `factors`?

Reply:
I was thinking diagview(A.factors) would be better

Reply:
Yes this is actually much faster than mine